src/commands/createNewProject/ProjectCreateStep/PowerShellProjectCreateStep.ts

```diff
@@ -54,9 +69,100 @@ export class PowerShellProjectCreateStep extends ScriptProjectCreateStep {
             await fse.writeFile(profileps1Path, profileps1);
         }
 
+        const majorVersion: number | undefined = await this.getLatestAzModuleMajorVersion(progress);
+        if (majorVersion !== undefined) {
+            progress.report({
+                message: localize('successfullyConnected', 'Successfully retrieved {0} information from PowerShell Gallery', this.azModuleName)
+            });
+        } else {
+            progress.report({
+                message: localize('failedToConnect', 'Failed to get {0} module version. Edit the requirements.psd1 file when the powershellgallery.com is accessible.', this.azModuleName)
+            });
+        }
+
         const requirementspsd1Path: string = path.join(context.projectPath, requirementspsd1FileName);
         if (await confirmOverwriteFile(requirementspsd1Path)) {
-            await fse.writeFile(requirementspsd1Path, requirementspsd1);
+            if (majorVersion !== undefined) {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1(majorVersion));
+            } else {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1Offine);
+            }
+        }
+    }
+
+    private async getLatestAzModuleMajorVersion(progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<number | undefined> {
+        progress.report({
+            message: localize('connecting', 'Connecting to PowerShell Gallery...')
+        });
+
+        const xmlResult: string | undefined = await this.getPSGalleryAzModuleInfo();
+        if (!xmlResult) {
+            return undefined;
+        }
+
+        const versionResult: string | undefined = await this.parseLatestAzModuleVersion(xmlResult);
+        if (!versionResult) {
+            return undefined;
+        }
+
+        try {
+            const [major]: string[] = versionResult.split('.');
+            return parseInt(major);
+        } catch {
+            return undefined;
+        }
+    }
+
+    private async getPSGalleryAzModuleInfo(): Promise<string | undefined> {
```

Comment:
Assuming you take my suggestion above, you don't need the try/catch in this function and you can change the return type to just `string`